contracts/multisig/src/worker_set.rs

```diff
@@ -1,15 +1,15 @@
-use std::collections::BTreeSet;
+use std::collections::BTreeMap;
 
 use crate::key::PublicKey;
-use crate::msg::Signer;
 use axelar_wasm_std::Participant;
 use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{HexBinary, Uint256};
 use sha3::{Digest, Keccak256};
 
 #[cw_serde]
 pub struct WorkerSet {
-    pub signers: BTreeSet<Signer>,
+    // An ordered map with the signer's address as the key, and the signer's weight and public key as the value.
+    pub signers: BTreeMap<String, (Uint256, PublicKey)>,
```

Comment:
don't make the key part of the value, if you make it a `BTreeMap<String, Signer>` you don't have to do all the mappings

Reply:
When you mention not making the key part of the value, are you referring to `PublicKey

Reply:
@cgorenflo When you mention not making the key part of the value, are you referring to `PublicKey` being part of the value?

The reason `Signer` is no longer the value is because it would cause the `address` field to be in both the `BTreeMap` key, and the value in `Signer.address`.
https://github.com/axelarnetwork/axelar-amplifier/pull/186#discussion_r1388482431

Reply:
`BTreeMap` is back to a `Signer` value for the following reason below as per @cgorenflo's suggestion.
When using maps, it's good practice to have the key only be used for looking up the value, rather than also being part of the value (and subsequently being used in the code's logic).
For example, in my code I often use the `address` (key) to rebuild the `Signer` struct, when it's instead better to directly store the `Signer` struct in the map for that key.
Another reason is because if I need to change the datatype of `address` later on, then at least the key still remains the same and does not need to be modified again.